### PR TITLE
always trigger a new completion popup on period

### PR DESCRIPTION
**Pull request by kasperpeulen**
(even if it is already active)

### CODE REVIEW

File: lib/playground.dart [hunk 1]
```diff
@@ -385,13 +382,11 @@ class Playground implements GistContainer {
           editor.execCommand("autocomplete");
         }
     } else if (context.focusedEditor == "html") {
-      if (options.getValueBool('autopopup_code_completion')) {
         // TODO: Autocompletion for attributes.
```
Comment (devoncarew) [inline lib/playground.dart]: indent?

File: lib/playground.dart [hunk 2]
```diff
@@ -364,17 +364,14 @@ class Playground implements GistContainer {
...
-    if (context.focusedEditor == 'dart') {
+    if (context.focusedEditor == 'dart' && editor.hasFocus) {
       if (e.keyCode == KeyCode.PERIOD) {
         editor.completionAutoInvoked = true;
         editor.execCommand("autocomplete");
       }
     }
-    if (!options.getValueBool('autopopup_code_completion')) {
+
+    if (!options.getValueBool('autopopup_code_completion') || _isCompletionActive || !editor.hasFocus) {
```
Comment (devoncarew) [inline lib/playground.dart]: long line?

File: lib/playground.dart (hunk 1 above)
Comment (kasperpeulen) [inline lib/playground.dart]: done

File: lib/playground.dart (hunk 2 above)
Comment (kasperpeulen) [inline lib/playground.dart]: done

**Comment by devoncarew** [comment]
lgtm!

**Pull request merged**